src/index/vocabulary/UnicodeVocabulary.h

```diff
@@ -5,8 +5,20 @@
 #ifndef QLEVER_SRC_INDEX_VOCABULARY_UNICODEVOCABULARY_H
 #define QLEVER_SRC_INDEX_VOCABULARY_UNICODEVOCABULARY_H
 
+#include <type_traits>
+
+#include "concepts/concepts.hpp"
+#include "index/vocabulary/PolymorphicVocabulary.h"
 #include "index/vocabulary/VocabularyTypes.h"
 
+// Only the `SplitVocabulary` currently needs a special handling for
+// `getPositionOfWord` (this includes the `PolymorphicVocabulary` which may
+// dynamically hold a `SplitVocabulary`)
+template <typename T>
+CPP_concept HasSpecialGetPositionOfWord =
+    std::is_same_v<T, PolymorphicVocabulary> ||
+    ad_utility::isInstantiation<T, SplitVocabulary>;
```

Comment:
As for the other place, 
make the explicit concept for the other vocabularies that don't have it currently, to make the misuse less likely.